packages/devtools_app/lib/src/screens/debugger/call_stack.dart

```diff
@@ -20,6 +20,8 @@ class CallStack extends StatefulWidget {
 
 class _CallStackState extends State<CallStack>
     with ProvidedControllerMixin<DebuggerController, CallStack> {
+  StackFrameAndSourcePosition? selectedFrame;
```

Comment:
it looks like this can get out of sync with `controller.selectedStackFrame` now that this widget is not responding to changes from this value listenable. Can we still pull the selected stack from from `controller.selectedStackFrame` so that the UI is always in sync with the controller?

Reply:
We can, though if we just do that the UI will seem unresponsive (see gif - the row is pink when it is tapped, and then turns grey when the `controller.selectedStackFrame` is updated). 

![delay in selecting stack frame](https://github.com/flutter/devtools/assets/21270878/8edab5ae-dc79-4539-9731-cac848ff50dd)

Explanation:
- we currently have a `postFrameCallback` in `codeview.dart` that will scroll to the selected line when the `selectedStackFrame` is updated
- we are waiting until after the script is successfully parsed and loaded to update the `selectedStackFrame` (not waiting was what caused the regression)
- therefore there is a lag after a user has clicked and before the `selectedStackFrame` is updated

I've updated it so that we are highlighting if either the row has just been clicked, or if the frame is selected. It means that for a short time we have two frames highlighted, but at least it seems responsive. WDYT?


![delay 2](https://github.com/flutter/devtools/assets/21270878/b91da2d0-1822-4d9f-b292-8a9e42b8862f)